compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ConstantTypeChecker.java

```diff
@@ -511,6 +513,36 @@ public void visit(BLangUnaryExpr unaryExpr, AnalyzerData data) {
         data.resultType = finiteType;
     }
 
+    @Override
+    public void visit(BLangStringTemplateLiteral stringTemplateLiteral, AnalyzerData data) {
+        StringBuilder resultString = new StringBuilder();
+        stringTemplateLiteral.exprs.forEach(expr -> {
+            BType exprType = checkConstExpr(expr, data);
+            if (exprType == symTable.semanticError) {
+                data.resultType = symTable.semanticError;
+                return;
```

Comment:
Here we only exit the lambda, not the `visit` method. Right? if `exprType` is `semanticError` we can set the `data.resultType` to `semanticError` and exit the visitor.